.github/workflows/main.yml

```diff
@@ -13,12 +13,25 @@ on:
 jobs:
   build:
     name: Test Runner
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v1
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
 
-      - name: Build Docker Image
-        run: docker build -f Dockerfile -t python-test-runner .
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4c0219f9ac95b02789c1075625400b2acbff50b1
+        with:
+          install: true
 
-      - name: Run Tests
-        run: docker run --entrypoint pytest python-test-runner -vv
+      - name: Build Docker image and store in cache
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825
+        with:
+          context: .
+          push: false
+          load: true
+          tags: exercism/python-test-runner
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Run Tests in Docker
+        run: bin/run-tests-in-docker.sh
```

Comment:
The main change is that running the tests is now a script, instead of being present only in the GHA workflow

Reply:
This tests the test runner with golden files....but we also have a script to test that all example/exemplar solutions for exercises pass all their tests.  We typically run that in the content repo on PR, but wondering if we should also do that here?  It adds a little time to the CI, but not much ... maybe 30-40s.

And, thinking about that - do we want a `run-tests-in-docker.sh` for the content repo that runs all the example and exemplar tests, or is it ok to leave it a Python script?